param/reactive.py

```diff
@@ -256,8 +258,12 @@ def map(self, func, /, *args, **kwargs):
         kwargs: mapping, optional
           A dictionary of keywords to pass to `func`.
         """
-        def apply(vs, *args, **kwargs):
-            return [func(v, *args, **kwargs) for v in vs]
+        if inspect.iscoroutinefunction(func):
```

Comment:
Should we raise an exception if a generator is used for maps? I don't think it makes much sense.

Reply:
Agreed, I almost went down the road of implementing it but it was just a little too crazy to be sensible.